RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -111,26 +118,120 @@ std::pair<std::vector<T>, std::vector<T>> categorise(
 
   return output;
 }
+template <typename T>
+std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> L1TJetsMatching<T>::categoriseVBFPlus2CentralJets(
+    const std::vector<T>& pfMatchedJets, double pt1, double pt2, double pt3, double Mjj) const {  //60, 30, 50, 500
+  std::tuple<std::vector<T>, std::vector<T>, std::vector<T>> output;
+  unsigned int i1 = 0;
+  unsigned int i2 = 0;
+
+  double mjj = 0;
+  if (pfMatchedJets.size() > 1) {
+    for (unsigned int i = 0; i < pfMatchedJets.size() - 1; i++) {
+      const T& myJet1 = (pfMatchedJets)[i];
+
+      for (unsigned int j = i + 1; j < pfMatchedJets.size(); j++) {
+        const T& myJet2 = (pfMatchedJets)[j];
+
+        const double mjj_test = (myJet1.p4() + myJet2.p4()).M();
```

Comment:
To improve performance, you could replace `M()` with `M2()` (even in another cleanup PR):
```suggestion
        const double mjj_test = (myJet1.p4() + myJet2.p4()).M();
```
```suggestion
        const double m2jj_test = (myJet1.p4() + myJet2.p4()).M2();
```